cmd/piecestore/main.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"sort"
+
+	"github.com/urfave/cli"
+	"github.com/zeebo/errs"
+	"storj.io/storj/pkg/piecestore"
+)
+
+var ArgError = errs.Class("argError")
+
+func main() {
+	app := cli.NewApp()
+
+	app.Name = "Piece Store CLI"
+	app.Usage = "Store data in hash folder structure"
+	app.Version = "1.0.0"
+
+	app.Flags = []cli.Flag{}
+
+	app.Commands = []cli.Command{
+		{
+			Name:      "store",
+			Aliases:   []string{"s"},
+			Usage:     "Store data by hash",
+			ArgsUsage: "[hash] [dataPath] [storeDir]",
+			Action: func(c *cli.Context) error {
+				if c.Args().Get(0) == "" {
+					return ArgError.New("Missing data Hash")
+				}
+
+				if c.Args().Get(1) == "" {
+					return ArgError.New("No input file specified")
+				}
+
+				if c.Args().Get(2) == "" {
+					return ArgError.New("No output directory specified")
+				}
+
+				file, err := os.Open(c.Args().Get(1))
+				if err != nil {
+					return err
+				}
+
+				// Close the file when we are done
+				defer file.Close()
+
+				fileInfo, err := os.Stat(c.Args().Get(1))
+				if err != nil {
+					return err
+				}
+
+				if fileInfo.IsDir() {
```

Comment:
you don't need to do this check. the file is definitely not a directory if os.Open worked

Reply:
https://golang.org/src/os/file.go?s=7402:7439#L23

Reply:
okay so after testing, i still don't think you need to do this check. you are right that the open will succeed but the first read will fail. the practical result is the same.